aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -487,24 +637,37 @@ struct cpu_scatter_gather_base_kernel {
         iter.for_each(loop, grain_size);
       }
     );
+    if (need_acc) {
+      self.copy_(buffer);
+    }
   }
 
   void operator()(const Tensor& self, int64_t dim,
     const Tensor& index, const Tensor& src,
     const std::string& method_name, ReduceMinimum& kernel_func) {
 
+    Tensor buffer;
+    bool need_acc = isReducedFloatingType(self.scalar_type());
```

Comment:
there are a couple of this section here, you can write an inline func to make the code more tight.